### PR TITLE
modernize

### PR DESCRIPTION
**Pull request by antis81**
Raise the minimum required CMake version to 3.5 (requires Sailfish SDK >= 2.1) to make use of Qt's modern CMake3 modules.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,33 +1,51 @@
 project(cmakesample CXX)
-cmake_minimum_required(VERSION 2.8.11)
+#cmake_minimum_required(VERSION 2.8.11)
```
Comment (pvuorela) [inline CMakeLists.txt]: Think we can just assume 3.x getting used already and remove commented out lines like this etc.

Comment (antis81) [reply CMakeLists.txt]: Ok.

File: qml/cmakesample_generic.qml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import QtQuick 2.0
+
+Rectangle {
+    width: 100
+    height: 100
+    color: "baf"
```
Comment (pvuorela) [inline qml/cmakesample_generic.qml]: If the app gets into loading qml, this will break it.

File: main.cpp [hunk 3]
```diff
@@ -52,3 +58,29 @@ int main(int argc, char *argv[])
     return SailfishApp::main(argc, argv);
 }
 
+#else
+// Example 2: use custom QQuickView
+int main(int argc, char *argv[])
+{
+    // FIXME: keeps crashing! -> provide a working example
```
Comment (pvuorela) [inline main.cpp]: The problem here is that Sailfish uses a booster (mapplauncherd). It preloads a process that instantiates QGuiApplication and QQuickView (or depends a bit on booster type) and when application gets actually launched, those preinstantiated things are set to load qml source. Desktop file has "X-Nemo-Application-Type=silica-qt5" which makes such trigger.

So what the app should do is export the main function and use mdeclarativecache like e.g. in https://github.com/sailfishos/sailfish-browser/blob/master/src/main.cpp

Comment (antis81) [reply main.cpp]: Thanks for this info! I will split the topic into another "future" branch then.

**Comment by antis81** [comment]
@pvuorela FYI: I decided to rebase the whole thing and split it into separate topics. Now it is better to review and merge. :)
Unfortunately we lost a piece of unrelated conversation along with the action. Ideally those comments would be still visible in Github (prbly even reassignable to #2) however not a big deal in this context.

P.S.: Github seems to assign a conversation to the "initial" commit of the PR. Now if the hash changes, it loses everything but the PR's title and description. Maybe I should open an issue to Microsoft to support such action in the future… :thinking:

File: qml/cmakesample_generic.qml (hunk 2 above)
Comment (antis81) [reply qml/cmakesample_generic.qml]: Done in #2

**Comment by antis81** [comment]
@pvuorela Can you please review again and prbly merge in case it is ok? Please let me know if this is complete bullshit. In case we need to support both CMake 2.x & 3.x versions I can think of providing 2 CMakeList.txt files - one CMakeLists.legacy.txt and one CMakeLists.txt (which would be the current by default).

My second PR builds on CMake3.x, stretching possibilities.

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,33 +1,38 @@
 project(cmakesample CXX)
-cmake_minimum_required(VERSION 2.8.11)
+cmake_minimum_required(VERSION 3.5)
 
-find_package(Qt5Qml REQUIRED)
+find_package (Qt5 COMPONENTS Core Network Qml Gui Quick REQUIRED)
 
 include(FindPkgConfig)
 pkg_search_module(SAILFISH sailfishapp REQUIRED)
+add_definitions(-DSAILFISHAPP)
```
Comment (pvuorela) [inline CMakeLists.txt]: This is not used yet, better not introduce before that.

**Comment by pvuorela** [comment]
Looks quite ok to me, one minor comment. Good idea to have separate PRs so different features can be integrated when ready.
Don't see reason to support old cmake, 3.x has been there long enough.

File: CMakeLists.txt (hunk 4 above)
Comment (antis81) [reply CMakeLists.txt]: done

**Comment by antis81** [comment]
Ok, I think it's ready. Thanks!

**Comment by pvuorela** [comment]
Looking good. For future cases can squash review fixes to main commit, no need to include intermediate steps in git history.

Thanks!

**Pull request merged**